### PR TITLE
SDIT-1208 Add alternative role to api/offenders/{offenderNo}/offender-restrictions endpoint

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java [hunk 1]
```diff
@@ -627,7 +627,7 @@ public OffenderContacts getOffenderContacts(
             @ApiResponse(responseCode = "404", description = "Requested resource not found.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
             @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
     @Operation(summary = "Gets the offender visit restrictions for a given offender using the latest booking", description = "Get offender visit restrictions by offender No")
-    @VerifyOffenderAccess(overrideRoles = {"SYSTEM_USER"})
+    @VerifyOffenderAccess(overrideRoles = {"SYSTEM_USER", "OFFENDER_RESTRICTIONS"})
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java:630]: Wasn't sure which role to add here.  There is a similar endpoint for offender-contacts which has OFFENDER_CONTACTS role as an alternative to SYSTEM_USER.  Usual VIEW_PRISONER_DATA seemed too general.

Comment (steverendell) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java:630]: Maybe use a 'visits' role as its related to that?

Comment (steverendell) [reply src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java:630]: Not that there seems to already be one at first glance so maybe this is fine

**Review by steverendell — APPROVED** [review]
lgtm

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**